### PR TITLE
merge ent-rep-ser-con into master

### PR DESCRIPTION
**Pull request by LorenzoGaetaniFinc**

### CODE REVIEW

File: src/main/java/com/fincons/taskmanager/config/AppConfig.java [hunk 1]
```diff
@@ -22,5 +25,18 @@ public ModelMapper modelMapperStandard(){
         ModelMapper modelMapper = new ModelMapper();
         return modelMapper;
     }
+    @Bean
+    public ModelMapper modelMapperForBoard(){
```
Comment (Simone-Puce) [inline src/main/java/com/fincons/taskmanager/config/AppConfig.java:29]: Why does a method that maps board has task and task dto in it ?

File: src/main/java/com/fincons/taskmanager/controller/AttachmentController.java [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+                    GenericResponse.error(
+                            rnfe.getMessage(),
+                            HttpStatus.NOT_FOUND
+                    )
+            );
+        }
+    }
+
+    @GetMapping(value = "${attachment.list}")
+    public ResponseEntity<GenericResponse<List<AttachmentDTO>>> getAllAttachment() {
```
Comment (Simone-Puce) [inline src/main/java/com/fincons/taskmanager/controller/AttachmentController.java:59]: get all attachement should be plural name

File: src/main/java/com/fincons/taskmanager/dto/TaskDTO.java [hunk 3]
```diff
@@ -21,5 +21,5 @@ public class TaskDTO {
     private String description;
     private List<UserDTO> users;
     private String boardCode;
-    private List<AttachmentDTO> attachmentDTOs;
+    private List<AttachmentDTO> attachments;
```
Comment (Simone-Puce) [inline src/main/java/com/fincons/taskmanager/dto/TaskDTO.java:24]: A list of attachamentDto should not be names attachement

**Review by Simone-Puce — CHANGES_REQUESTED** [review]

File: src/main/java/com/fincons/taskmanager/controller/AttachmentController.java (hunk 2 above)
Comment (LorenzoGaetaniFinc) [reply src/main/java/com/fincons/taskmanager/controller/AttachmentController.java:59]: Resolve

File: src/main/java/com/fincons/taskmanager/dto/TaskDTO.java (hunk 3 above)
Comment (LorenzoGaetaniFinc) [reply src/main/java/com/fincons/taskmanager/dto/TaskDTO.java:24]: The Model Mapper library require the same name, for mapping correctly

File: src/main/java/com/fincons/taskmanager/config/AppConfig.java (hunk 1 above)
Comment (LorenzoGaetaniFinc) [reply src/main/java/com/fincons/taskmanager/config/AppConfig.java:29]: Board have a daughter Task, and Task have a daughter Attachment, we don't wont mapping attachment when we find Board. We need Attachment only when we mapping Task

**Review by Simone-Puce — APPROVED** [review]

**Pull request merged**